proto/valorem/trade/v1/sq.proto

```diff
@@ -0,0 +1,47 @@
+syntax = "proto3";
+
+import "types.proto";
+
+package valorem.trade.v1;
+
+/**
+ * Soft Quotes (SQ) services and related message formats.
+ */
+
+message SoftQuoteRequest {
+    /** The exercise token address */
+    H160 exercise = 1;
+
+    /** The underlying token address */
+    H160 underlying_asset = 2;
+
+    /** The amount of options the Taker is interested in purchasing */
+    H256 amount = 3;
+
+    /** A request by the Taker to the Maker, i.e. if the request is Buy the Taker wants to buy the option from the
+     * Maker, whereas Sell is the Taker wanting to sell to the Maker.
+     */
+    Action action = 4;
+
+    /** The strike prices (in USD) the Taker wishes to get a soft quote on */
+    repeated float strike = 5;
+}
+
+message SoftQuoteResponse {
+    /** The prices (in USD) the Maker would approximately charge. The order of the prices is in the same order
+     * as the requested strikes.
+     */
+    repeated float price = 1;
+}
+
+service SoftQuotes {
+    /** Request soft quotes from makers via a single SoftQuoteRequest message and receive a stream of SoftQuoteResponse
+     * messages for use by gRPC-web clients.
+     */
+    rpc SoftQuoteWebTaker (SoftQuoteRequest) returns (stream SoftQuoteResponse);

```

Comment:
Might we make this a matrix in some way, where the taker can request a matrix of soft quotes and get a stream of responses, and where the stream would denote the maker so that the user could then pass in the maker to the hard RFQ or that we could marry the soft quote to the hard quote in API internals somehow?

Reply:
So we can make the messages a message of messages to allow for many assets/strike soft quotes and add in the maker address as a part of the response so the user know who the maker is - we would need to edit the rfq message to include an optional maker address if we wanted to add in functionality allowing the taker to specify a maker to use.